src/test/obj_ulog_size/obj_ulog_size.c

```diff
@@ -256,63 +306,78 @@ do_tx_max_alloc_prealloc_nested(PMEMobjpool *pop)
 	} TX_END
 
 	UT_ASSERTeq(D_RO(obj1)->value, TEST_VALUE_2);
-	free_pool(oid);
+	free_pool(oids, noids);
 }
 
+/*
+ * do_tx_max_alloc_prealloc_snap_multi -- appending of many buffers
+ *		in one transaction
+ */
 static void
 do_tx_max_alloc_prealloc_snap_multi(PMEMobjpool *pop)
 {
-	UT_OUT("prealloc_snap_multi");
-	PMEMoid oid[MAX_OBJECTS];
+	UT_OUT("do_tx_max_alloc_prealloc_snap_multi");
+	PMEMoid *oids = NULL;
+	size_t noids = 0;
 
-	fill_pool(pop, oid);
+	oids = fill_pool(pop, &noids);
 
 	/* pool is full now, so let's try to snapshot first object */
-	size_t snap_size0 = pmemobj_alloc_usable_size(oid[0]);
-	void *addr0 = pmemobj_direct(oid[0]);
-	size_t snap_size1 = pmemobj_alloc_usable_size(oid[1]);
-	void *addr1 = pmemobj_direct(oid[1]);
-	size_t snap_size2 = pmemobj_alloc_usable_size(oid[2]);
-	void *addr2 = pmemobj_direct(oid[2]);
+	size_t snap_size0 =
+		pmemobj_alloc_usable_size(oids[PREALLOCATED_OBJECT]);
+	void *addr0 = pmemobj_direct(oids[PREALLOCATED_OBJECT]);
+	size_t snap_size1 =
+		pmemobj_alloc_usable_size(oids[PREALLOCATED_OBJECT + 1]);
+	void *addr1 = pmemobj_direct(oids[PREALLOCATED_OBJECT + 1]);
+	size_t snap_size2 =
+		pmemobj_alloc_usable_size(oids[PREALLOCATED_OBJECT + 2]);
+	void *addr2 = pmemobj_direct(oids[PREALLOCATED_OBJECT + 2]);
 
 	errno = 0;
 	TX_BEGIN(pop) {
-		/* abort expected - cannot allocate memory */
 		pmemobj_tx_log_append_buffer(
 			TX_LOG_TYPE_SNAPSHOT, addr0, snap_size0);
 		pmemobj_tx_log_append_buffer(
 			TX_LOG_TYPE_SNAPSHOT, addr1, snap_size1);
 		pmemobj_tx_log_append_buffer(
 			TX_LOG_TYPE_SNAPSHOT, addr2, snap_size2);
 
-		pmemobj_tx_add_range(oid[0], 0, snap_size0);
-		pmemobj_tx_add_range(oid[1], 0, snap_size1);
-		pmemobj_tx_add_range(oid[2], 0, snap_size2);
+		pmemobj_tx_add_range(oids[SNAPSHOT_OBJECT + 3], 0, snap_size0);
+		pmemobj_tx_add_range(oids[SNAPSHOT_OBJECT + 4], 0, snap_size1);
+		pmemobj_tx_add_range(oids[SNAPSHOT_OBJECT + 5], 0, snap_size2);
```

Comment:
If you need more than one object for"preallocated" please set "snapshot" far enough you do not need the index-magic. e.g.:

```c
#define LOG_BUFFER 0
#define LOG_BUFFER_NUM 10
#define RANGE (LOG_BUFFER + LOG_BUFFER_NUM)
#define RANGE_NUM 10
#define MIN_NOIDS (RANGE + RANGE_NUM)

/* you have to verify you have enough oids allocated ;-) */
oids = fill_pool(pop, &noids);
UT_ASSERT(noids >=MIN_NOIDS);
```